tests/src/test-agent.ts

```diff
@@ -146,14 +146,55 @@ test("Spawn with agent", async (t) => {
 
   // Result should respond to ping.
   const result = await axios.get(`http://${address}`)
+  console.log(result)
   t.is(result.status, 200)
   t.is(result.data, "Hello World!")
-
+  
   // Status should update to swept after ~10 seconds.
   t.is("Swept", (await backendStatusSubscription.next())[0].state)
   t.is("Swept", (await t.context.db.getBackend(backendId)).state)
 })
 
+test("stats are acquired", async (t) => {
+  const backendId = generateId()
+  const natsPort = await t.context.docker.runNats()
+  await sleep(100)
+  const nats = await connect({ port: natsPort, token: "mytoken" })
+  t.context.runner.runAgent(natsPort)
+
+  await expectMessage(t, nats, "drone.register", {
+    cluster: "mydomain.test",
+    ip: "123.12.1.123",
+  }, {
+    Success: {
+      drone_id: 1,
+    },
+  })
+
+  await sleep(100)
+
+  // Spawn request.
+  const request: SpawnRequest = {
+    image: TEST_IMAGE,
+    backend_id: backendId,
+    max_idle_secs: 40,
+    env: {
+      PORT: "8080",
+    },
+    metadata: {},
+  }
+  expectResponse(t, nats, "drone.1.spawn", request, true)
+
+  const statsStatusSubsription = 
+    new NatsMessageIterator<unknown>(
+      nats.subscribe(`backend.${backendId}.stats`, { timeout : 10000 })
+  )
+  t.deepEqual(["cpu_used", "mem_used", "disk_used"],
+    Object.keys((await statsStatusSubsription.next())[0]))
```

Comment:
This enforces a key order which is stricter than we care about (for the purposes of our interfaces, objects should not have ordered keys).

Maybe we should individually test that each of these are numbers strictly greater than 0 and less than 1?